src/plugins/data/public/ui/query_string_input/query_string_input.tsx

```diff
@@ -257,16 +264,19 @@ export class QueryStringInputUI extends Component<Props, State> {
 
       switch (event.keyCode) {
         case KEY_CODES.DOWN:
-          event.preventDefault();
           if (isSuggestionsVisible && index !== null) {
+            event.preventDefault();
             this.incrementIndex(index);
-          } else {
+            // Note to engineers. `isSuggestionVisible` does not mean the suggestions are visible.
+            // This should likely be fixed, it's more that suggestions can be shown.
```

Comment:
@XavierM @spong 

I HIGHLY suggest taking a look at the state changes in this code when you run through here. At best this state is improperly named. `isSuggestionsVisible` is more accurately "mightHaveSuggestions". There are times when it is `true` when the `index` is actually set.

My logic works, but I wouldn't be suprised if the next person who reads this code will likely run into an hour of headscratching like I did trying to figure out what is wrong.

I did not change anything here related to the state change, because this is a big codebase and I didn't want to muck it up, but one of you should check it out. I could only diagnose the problem by adding a bunch of console logs and seeing how the state changed in various `keyEvent`s